src/storage/txn/store.rs

```diff
@@ -232,50 +222,46 @@ impl Store for FixtureStore {
             }
         });
 
-        let mut vec: Vec<_> = self
-            .data
-            .range((lower, upper))
-            .map(|(k, v)| {
-                let owned_k = k.clone();
-                let owned_v = if key_only {
-                    match v {
-                        Ok(_v) => Ok(vec![]),
-                        Err(e) => Err(e.maybe_clone().unwrap()),
-                    }
-                } else {
-                    match v {
-                        Ok(v) => Ok(v.clone()),
-                        Err(e) => Err(e.maybe_clone().unwrap()),
-                    }
-                };
-                (owned_k, owned_v)
-            })
-            .collect();
-
-        if desc {
-            vec.reverse();
-        }
-
         Ok(FixtureStoreScanner {
-            // TODO: Remove clone when GATs is available. See rust-lang/rfcs#1598.
-            data: vec.into_iter(),
+            key_only,
+            desc,
+            handle: OwningHandle::new_with_fn(self.data.clone(), |m| unsafe {
+                Box::new((*m).range((lower, upper)))
```

Comment:
Having thought about this a bit, I think this might be unsound: once the original `Arc` is destroyed, the one in the `OwningHandle` will be the only reference and so `try_unwrap` on the internal `Arc` (which is available via `as_owner`) will succeed, then the map can be mutated which could cause the internal data to move and thus the `Range` to be pointing at a freed pointer.

I think the easy solution is to add_ref the `Arc` before putting it in the `OwningHandle`, and deref'ing when the handle is dropped. Alternatively, we could write our own version of `OwningHandle` specialised for `Arc`s (I would prefer this since the owning_ref crate doesn't seem to be maintained, we might be able to use `Pin` to make things safer too).